tensorflow_lite_support/ios/task/processor/sources/TFLSegmentationResult+Helpers.m

```diff
@@ -0,0 +1,77 @@
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================*/
+#import "tensorflow_lite_support/ios/task/processor/sources/TFLSegmentationResult+Helpers.h"
+
+@implementation TFLSegmentationResult (Helpers)
+
++ (TFLSegmentationResult *)segmentationResultWithCResult:
+    (TfLiteSegmentationResult *)cSegmentationResult {
+  if (cSegmentationResult == nil) return nil;
+
+  NSMutableArray *segmentations = [[NSMutableArray alloc] init];
+  for (int i = 0; i < cSegmentationResult->size; i++) {
+    TfLiteSegmentation cSegmentation = cSegmentationResult->segmentations[i];
+    NSMutableArray *coloredLabels = [[NSMutableArray alloc] init];
+    for (int j = 0; j < cSegmentation.colored_labels_size; j++) {
+      TfLiteColoredLabel cColoredLabel = cSegmentation.colored_labels[j];
+
+      TFLColoredLabel *coloredLabel = [[TFLColoredLabel alloc] init];
+      coloredLabel.r = cColoredLabel.r;
+      coloredLabel.g = cColoredLabel.g;
+      coloredLabel.b = cColoredLabel.b;
+
+      if (cColoredLabel.display_name != nil) {
+        coloredLabel.displayName = [NSString stringWithCString:cColoredLabel.display_name
+                                                      encoding:NSUTF8StringEncoding];
+      }
+
+      if (cColoredLabel.label != nil) {
+        coloredLabel.label = [NSString stringWithCString:cColoredLabel.label
+                                                encoding:NSUTF8StringEncoding];
+      }
+
+      [coloredLabels addObject:coloredLabel];
+    }
+
+    TFLSegmentation *segmentation = [[TFLSegmentation alloc] init];
+    segmentation.coloredLabels = coloredLabels;
+
+    segmentation.width = cSegmentation.width;
+    segmentation.height = cSegmentation.height;
+    segmentation.coloredLabelsSize = cSegmentation.colored_labels_size;
+
+    if (cSegmentation.confidence_masks) {
+      segmentation.confidenceMasks = malloc(segmentation.coloredLabelsSize * sizeof(float *));
+
+      for (int i = 0; i < segmentation.coloredLabelsSize; i++) {
+        segmentation.confidenceMasks[i] =
+            malloc(segmentation.width * segmentation.height * sizeof(float));
+        memcpy(segmentation.confidenceMasks[i], cSegmentation.confidence_masks[i],
+               segmentation.width * segmentation.height);
+      }
+    } else if (cSegmentation.category_mask) {
+      segmentation.categoryMask = malloc(segmentation.width * segmentation.height * sizeof(int));
+      memcpy(segmentation.categoryMask, cSegmentation.category_mask,
```

Comment:
@priankakariatyml As you need to copy from int to NSUInteger, you'll need to create a for loop to cast each element from int to NSUInteger.

Reply:
Fixed this and few other issues. Please pull the latest changes.